tensorboard/plugins/custom_scalar/layout.py

```diff
@@ -0,0 +1,51 @@
+# Copyright 2017 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Contains logic related to laying out the custom scalars dashboard.
+"""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import os
+
+import tensorflow as tf
+
+CONFIG_FILE = 'custom_scalars_config.pbtxt'
+
+def set_layout(logdir, scalars_layout):
```

Comment:
Better to pass some SummaryWriter-esque object rather than a logdir, since the SummaryWriter is the canonical point by which the summary data is exported.

Reasons:
1. One often keeps access to a SummaryWriter when writing code that deals with summaries, but the logdir might be in some outer scope
2. In the future, we may switch to e.g a DatabseWriter or RemoteServerWriter and in these cases there isn't meaningfully a "logdir". It would be better to implement a consistent interface across all our writers that can handle this, rather than privilging the logdir concept.